operator/validator/metadata/sync_batch.go

```diff
@@ -0,0 +1,51 @@
+package metadata
+
+import (
+	"github.com/attestantio/go-eth2-client/spec/phase0"
+	spectypes "github.com/ssvlabs/ssv-spec/types"
+
+	ssvtypes "github.com/ssvlabs/ssv/protocol/v2/types"
+)
+
+type SyncBatch struct {
+	SharesBefore []*ssvtypes.SSVShare
```

Comment:
Do we still use the fields outside `SyncBatch`'s methods? Should we do something like this instead?

```go
func NewSyncBatch(before, after []*ssvtypes.SSVShare, epoch phase0.epoch) *SyncBatch {
    attesting, slashed, exited := detectValidatorStateChanges(before, after, epoch)
    return &SyncBatch{
        Attesting: attesting,
        Slashed: slashed,
        Exited: exited,
    }
}
```